libraries/chain/controller.cpp

```diff
@@ -354,14 +397,25 @@ struct controller_impl {
          emit( self.accepted_block_header, pending->_pending_block_state );
          head = fork_db.head();
          FC_ASSERT( new_bsp == head, "committed block did not become the new head in fork database" );
+
+      }
+
+      if( !replaying ) {
+         auto itr = unconfirmed_blocks.find<unconfirmed_block_object,by_num>(head->block_num );
+         if( !itr ) {
```

Comment:
Can't we assert `!itr`? It seems like it would be a sign of a bug if we commit a block that already exists in the `unconfirmed_blocks`.